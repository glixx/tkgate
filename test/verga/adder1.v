module top;
  wire s,co;
  reg a,b,ci;

  adder a1(s,co,a,b,ci);

  initial
    begin
	$monitor("%d: {%h,%h} = %h + %h + %h",$time,co,s,a,b,ci);

	#2 a = 1'b0; b = 1'b0; ci = 1'b0;
	#2 a = 1'b0; b = 1'b0; ci = 1'b1;
	#2 a = 1'b0; b = 1'b1; ci = 1'b0;
	#2 a = 1'b0; b = 1'b1; ci = 1'b1;
	#2 a = 1'b1; b = 1'b0; ci = 1'b0;
	#2 a = 1'b1; b = 1'b0; ci = 1'b1;
	#2 a = 1'b1; b = 1'b1; ci = 1'b0;
	#2 a = 1'b1; b = 1'b1; ci = 1'b1;
    end

endmodule

module adder(s,co,a,b,ci);
  output s,co;
  input a,b,ci;
  wire x1,w1,w2,w3;

  xor (x1,a,b);
  xor (s,x1,ci);

  and (w1,a,b);
  and (w2,a,ci);
  and (w3,b,ci);

  or (co, w1, w2, w3);

endmodule
