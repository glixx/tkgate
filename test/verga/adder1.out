0: {x,x} = x + x + x
2: {0,0} = 0 + 0 + 0
4: {0,1} = 0 + 0 + 1
6: {0,1} = 0 + 1 + 0
8: {1,0} = 0 + 1 + 1
10: {0,1} = 1 + 0 + 0
12: {1,0} = 1 + 0 + 1
14: {1,0} = 1 + 1 + 0
16: {1,1} = 1 + 1 + 1
